### ISSUE TITLE
Question: How can I remove inline styles on paste?

### ISSUE THREAD

**Issue by jessegavin** [bevacqua/woofmark#49]
I want to be able to use woofmark as a wysiwyg editor for markdown only. 

1. I never want my users to be able to switch to html mode (I know how to do this). 
2. I would like to automatically strip out any inline styles when a user pastes rich text into the editor while in wysiwyg mode. ( **don't** know how to do this).

How can I accomplish the second item?

I created a code pen to illustrate the non-desirable behavior. http://codepen.io/jessegavin/pen/mRyOJb?editors=0010

Thanks so much. This is one of the coolest implementations I have seen for a wysiwyg editor.

**Comment by jywarren** [comment]
My understanding is that they'll be stripped out when the content is
converted into markdown, but that's not triggered on paste. I think it
could be a good idea to add some kind of filter to pasted content, or as
you've shown, it tends to persist in the editor in WYSIWYG mode until
something triggers the entire post body to be parsed.

**Comment by jessegavin** [comment]
@jywarren that is correct. I can get the desired result if I paste inline-styled content into the wysiwyg mode, then switch to markdown, then switch back.

I want some way to skip the step of having to manually switch to markdown mode and back.

**Comment by jywarren** [comment]
I've tried implementing a triggering of parsing back and forth on a paste
event, but it's not ideal -- for one, it sort of hiccups, and I forget,
perhaps the insertion point was moved or lost? In any case you can give it
a try -- woofmark uses the crossvent event library, also by @bevacqua. It's
pretty easy to use!

**Comment by jessegavin** [comment]
Ok, I have come up with a "solution" which works for _my_ particular needs, but it doesn't feel clean at all.

And, because it uses `setTimeout()` there's a visual "blip" when the pasted content is displayed with inline styles, then immediately they are removed.

```
var editor = woofmark(...);

editor.editable.addEventListener('paste', function(e) {
  setTimeout(function() {
    var els = editor.editable.querySelectorAll('*'), i;
    for (i = 0; i < els.length; i++) {
      els[i].removeAttribute('style');
    }
  },0);
}, false);
```

**Comment by jywarren** [comment]
Hmm, do you think it'd be possible to intercept the pasted content and
preventDefault() to stop it getting to the textarea at all? Then insert it
via the "chunks" object, as shown in the code example here:

https://github.com/publiclab/PublicLab.Editor/issues/85#issuecomment-271722433

Just an idea!
